File: bench/loop.c
#include <stdio.h>
long inner(long acc, long n){ while(n){ acc=(acc+n*n)%2147483647L; n--; } return acc; }
long outer(long acc, long k){ while(k){ acc+=inner(k,50000); k--; } return acc; }
int main(void){ printf("%ld\n", outer(0,4000)); return 0; }

File: bench/loop.rs
fn inner(mut acc:i64, mut n:i64)->i64{ while n!=0 { acc=(acc+n.wrapping_mul(n))%2147483647; n-=1; } acc }
fn outer(mut acc:i64, mut k:i64)->i64{ while k!=0 { acc=acc.wrapping_add(inner(k,50000)); k-=1; } acc }
fn main(){ println!("{}", outer(0,4000)); }
